op-batcher/batcher/driver.go

```diff
@@ -318,6 +318,11 @@ func (l *BatchSubmitter) publishStateToL1(queue *txmgr.Queue[txData], receiptsCh
 				}
 				return
 			}
+			// errors from Txmgr are only seen by the goroutine that sends the tx
+			// so we need to check if the txmgr is closed here to avoid retrying forever
+			if l.Txmgr.IsClosed() {
+				return
+			}
```

Comment:
I think we can even move this up in the loop, before the `publishTxToL1` call, saving one error log of an expected situation. We might still do an `Info` log here that `publishTxToL1` is returning because the txmgr is closed.